File: fqnews/core/gfwlist/gen.pl
#!/usr/bin/env perl
## ArchLinux install package via pacman: perl-net-cidr-lite
use strict;
use warnings;
use Net::CIDR::Lite;
my $cidr = Net::CIDR::Lite->new;
while (my $line=<>) {
    $cidr->add($line);
}
foreach my $line( @{$cidr->list} ) {
    print "<item>$line</item>\n";
}

File: fqnews/core/gfwlist/gen.py
#!/usr/bin/python
# -*- encoding: utf8 -*-

import sys

import IPy


def main():
    china_list_set = IPy.IPSet()
    for line in sys.stdin:
        china_list_set.add(IPy.IP(line))

    # 输出结果
    for ip in china_list_set:
        print '<item>' + str(ip) + '</item>'


if __name__ == "__main__":
    main()
